core/src/components/progress-bar/progress-bar.scss

```diff
@@ -0,0 +1,232 @@
+@import "../../themes/ionic.globals";
+// Progress bar
+// --------------------------------------------------
+:host {
+    /**
+     * @prop --buffer-background: Background of the buffer bar
+     * @prop --buffer-size: Size of the buffer circles
+     * @prop --buffer-speed: Speed of the buffer circles
+     */
+    --buffer-background: rgba(0, 0, 0, .30);
+    --buffer-size: 10px;
+    --buffer-speed: 450ms;
+    display: block;
+
+    width: 100%;
+    height: 4px;
+
+    background: rgba(0, 0, 0, .12);
+
+    contain: strict;
+    overflow: hidden;
+}
+
+:host(.ion-color) {
+    --buffer-background: rgba(#{current-color(base-rgb)}, .30);
+    --buffer-color: #{current-color(base)};
+    background: rgba(#{current-color(base-rgb)}, .12);
+    color: #{current-color(base)};
+}
+
+// Default progress bar
+.progress {
+    @include transform-origin(start, top);
+    position: absolute;
+
+    width: 100%;
+    height: 100%;
+
+    transition: transform 150ms linear;
+}
+
+// Progress bar background
+:host(.buffer),
+:host(.reversed),
+:host(.determinate) {
+    .progress {
+        background: currentColor;
+    }
+}
+
+// On reversed the bar starts on the right
+:host(.reversed) {
+    .progress {
+        @include transform-origin(end, top);
+    }
+}
+
+// For indeterminate and query seperate bars will be needed
+// to add the MD based animation
+:host(.indeterminate),
```

Comment:
not needed

Reply:
.primary-bar and friends are only used when the type is query or indeterminate anyway, it does not need to be wrapped around `:host(.indeterminate),`